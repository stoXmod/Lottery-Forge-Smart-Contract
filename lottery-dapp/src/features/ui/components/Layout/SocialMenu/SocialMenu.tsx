import React, { ReactNode } from 'react';

import {
  chakra,
  Stack,
  useColorModeValue,
  VisuallyHidden,
} from '@chakra-ui/react';
import { FaGithub, FaReadme } from 'react-icons/fa';

const SocialButton = ({
  children,
  label,
  href,
}: {
  children: ReactNode;
  label: string;
  href: string;
}) => {
  return (
    <chakra.button
      bg={useColorModeValue('blackAlpha.100', 'whiteAlpha.100')}
      rounded="full"
      role="button"
      name={label}
      w={8}
      h={8}
      cursor="pointer"
      as="a"
      href={href}
      target="_blank"
      display="inline-flex"
      alignItems="center"
      justifyContent="center"
      transition="background 0.3s ease"
      _hover={{
        bg: useColorModeValue('blackAlpha.200', 'whiteAlpha.200'),
      }}
    >
      <VisuallyHidden>{label}</VisuallyHidden>
      {children}
    </chakra.button>
  );
};

export const SocialMenu: React.FC = React.memo(() => {
  return (
    <Stack direction="row" spacing={6}>
      <SocialButton
        label="GitHub"
        href="https://github.com/stoXmod/Lottery-System-Smart-Contract.git"
        key="GitHub"
      >
        <FaGithub />
      </SocialButton>
      <SocialButton
        label="Readme"
        href="https://huseyindeniz.github.io/react-dapp-template-documentation/"
        key="Documentation"
      >
        <FaReadme />
      </SocialButton>
    </Stack>
  );
});
