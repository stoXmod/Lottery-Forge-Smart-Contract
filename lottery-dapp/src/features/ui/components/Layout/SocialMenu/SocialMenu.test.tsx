import { composeStories } from '@storybook/react';
import { render, act } from '@testing-library/react';

import * as stories from './SocialMenu.stories';

describe('Feature: UI', () => {
  describe('Component: Layout/SocialMenu', () => {
    const { Default } = composeStories(stories);
    describe('Scenario: Default', () => {
      it('renders a profile button', async () => {
        // Arrange + Act
        const { asFragment, getByRole } = await act(async () =>
          render(<Default {...Default.args} />)
        );
        // Assert
        expect(getByRole('button', { name: 'GitHub' })).toBeInTheDocument();
        expect(getByRole('button', { name: 'Readme' })).toBeInTheDocument();
        expect(asFragment).toMatchSnapshot();
      });
    });
  });
});
