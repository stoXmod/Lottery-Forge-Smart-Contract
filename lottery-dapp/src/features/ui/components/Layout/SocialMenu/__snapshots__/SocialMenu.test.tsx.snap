// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: UI Component: Layout/SocialMenu Scenario: Default renders a profile button 1`] = `[Function]`;
